lib/OcrProcessors/PdfOcrProcessor.php

```diff
@@ -24,194 +24,30 @@
-					$splitted[$i] = $content;
-				} finally {
-					$onePageFpdiWrapper->Close();
-					$onePageFpdiWrapper->closeStreams();
-				}
-			}
-		} finally {
-			if (isset($fpdiWrapper)) {
-				$fpdiWrapper->Close();
-				$fpdiWrapper->closeStreams();
-			}
-		}
-
-		return $splitted;
-	}
-
-	/**
-	 * Process each PDF page with ocr algorithm except the pages which already
-	 * contain a text layer.
-	 */
-	private function ocrPages(array &$splittedPdfPages, array $pagesTextInfo) : void {
-		foreach ($splittedPdfPages as $i => $onePagePdf) {
-			// Skip pages containing text
-			if ($pagesTextInfo[$i] === true) {
-				continue;
-			}
-
-			try {
-				// Use Imagick to convert the pdf page to png
-				$img = $this->wrapperFactory->createImagick();
-				$img->setOption('density', '300');
-				$img->readImageBlob($onePagePdf);
-				$img->setImageFormat("png");
-
-				$ocrPdf = $this->processSinglePageImagick($img);
-
-				// Take original page format
-				$original = $this->wrapperFactory->createFpdi($onePagePdf);
-				$pageId = $original->import(1);
-				$originalSize = $original->getTemplatesize($pageId);
-
-				// Import single PDF page with ocr layer
-				$withOcr = $this->wrapperFactory->createFpdi($ocrPdf);
-				$pageIdOcr = $withOcr->import(1);
-				$withOcr->AddPage($originalSize['orientation'], $originalSize);
-				$withOcr->useImportedPage($pageIdOcr, 0, 0, $originalSize['width'], $originalSize['height'], false);
-
-				// Overwrite original page with scanned one
-				$splittedPdfPages[$i] = $withOcr->Output(null, "S");
-			} finally {
-				if (isset($img)) {
-					$img->destroy();
-				}
-				if (isset($original)) {
-					$original->Close();
-					$original->closeStreams();
-				}
-				if (isset($withOcr)) {
-					$withOcr->Close();
-					$withOcr->closeStreams();
-				}
-			}
-		}
-	}
-
-	private function processSinglePageImagick(IImagick $imagick) : string {
-		$data = $imagick->getImageBlob();
-		$size = $imagick->getImageLength();
-
-		// Use Tesseract for ocr and converting image back to pdf
-		$singlePagePdf = $this->tesseract
-			->lang(['deu', 'eng']) // TODO make configurable?
-			->imageData($data, $size)
-			->configFile('pdf')
-			->run();
-
-		return $singlePagePdf;
-	}
-
-	/**
-	 * Merges single page PDF array into one output PDF.
-	 */
-	private function mergePdf(array &$splitted) : string {
-		try {
-			$outputPdf = $this->wrapperFactory->createFpdi();
-
-			foreach ($splitted as $i => $onePageOcrPdf) {
-				$outputPdf->setContent($onePageOcrPdf);
-				$pageId = $outputPdf->import(1);
-				$s = $outputPdf->getTemplatesize($pageId);
-				$outputPdf->AddPage($s['orientation'], $s);
-				$outputPdf->useImportedPage($pageId);
-			}
-
-			$outputPdfContent = $outputPdf->Output(null, "S");
-			return $outputPdfContent;
-		} finally {
-			if (isset($outputPdf)) {
-				$outputPdf->Close();
-				$outputPdf->closeStreams();
-			}
+		$this->command
+			->setCommand("ocrmypdf --skip-text -q - - | cat")
```

Comment:
Please replace the `--skip-text` parameter of `OCRmyPDF` with the `--redo-ocr` parameter, otherwise images embedded into a text page are ignored for any OCR processing. In addition to that, the `--redo-ocr` parameter avoids a double insertion of invisible text (as OCR) if a PDF document is processed twice (due to an Nextcloud file update/modification event).

```suggestion
			->setCommand("ocrmypdf --redo-ocr -q - - | cat")
```